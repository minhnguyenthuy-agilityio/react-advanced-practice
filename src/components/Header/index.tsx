import { Avatar, Box, Flex, Icon, Text } from "@chakra-ui/react";

import { SearchBox } from "@/components";

import { USER } from "@/mocks";

import { ArrowDownIcon } from "@/assets/icons";

export const Header = () => {
  return (
    <Flex
      as="header"
      borderBottom="1px solid"
      borderColor="primary.400"
      px="55px"
      py="25px"
      justifyContent="space-between"
      alignItems="center"
    >
      <SearchBox />
      <Flex alignItems="center">
        <Box mr="20px" textAlign="right">
          <Text color="primary.700">{USER.name}</Text>
          <Text size="sm">{USER.country}</Text>
        </Box>
        <Avatar mr="10px" w="38px" h="38px" src={USER.avatar} />
        <Icon as={ArrowDownIcon} />
      </Flex>
    </Flex>
  );
};
