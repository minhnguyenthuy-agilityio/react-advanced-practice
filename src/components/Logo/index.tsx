import { Link as ChakraLink, Heading, Icon, Text } from "@chakra-ui/react";

import { Link as ReactRouterLink } from "react-router-dom";

import { LogoIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

export const Logo = () => (
  <Heading
    as="h1"
    py="35px"
    pl="20px"
    alignItems="center"
    borderBottom="solid 1px "
    borderColor="primary.400"
  >
    <ChakraLink
      display="flex"
      as={ReactRouterLink}
      to={ROUTES.TASKS_PROJECT_DEFAULT}
      _hover={{
        textDecoration: "none",
      }}
    >
      <Icon mr="8px" w="24px" h="24px" as={LogoIcon} />
      <Text color="primary.700" size="xm" fontWeight="semibold">
        Project M.
      </Text>
    </ChakraLink>
  </Heading>
);
