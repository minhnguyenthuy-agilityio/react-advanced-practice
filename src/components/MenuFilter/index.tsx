import { FC, memo } from "react";

import {
  Button,
  Icon,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
} from "@chakra-ui/react";

import { ArrowDownFilterIcon } from "@/assets/icons";

interface MenuFilterProps {
  buttonName: string;
  options: string[];
  icon?: FC<React.SVGProps<SVGSVGElement>>;
}

export const MenuFilter = memo(
  ({ buttonName, options, icon }: MenuFilterProps) => {
    return (
      <Menu>
        <MenuButton
          as={Button}
          variant="outline"
          leftIcon={icon && <Icon data-testid="icon-button-menu" as={icon} />}
          rightIcon={<Icon as={ArrowDownFilterIcon} />}
          w="122px"
          maxH="40px"
          fontWeight="medium"
        >
          {buttonName}
        </MenuButton>
        <MenuList>
          {options?.map((option, index) => (
            <MenuItem
              data-testid="menu-item"
              key={`${option}-${index}`}
              _hover={{
                bgColor: "secondary.200",
              }}
            >
              {option}
            </MenuItem>
          ))}
        </MenuList>
      </Menu>
    );
  },
);
