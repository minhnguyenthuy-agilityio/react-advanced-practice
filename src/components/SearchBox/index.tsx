import { Icon, Input, InputGroup, InputLeftElement } from "@chakra-ui/react";

import { SearchIcon } from "@/assets/icons";

export const SearchBox = () => {
  return (
    <InputGroup bg="primary.200" borderRadius="6px" maxW="417px">
      <InputLeftElement left="3" top="10%" fontSize="sm">
        <Icon as={SearchIcon} />
      </InputLeftElement>
      <Input
        pl="55px"
        h="full"
        placeholder="Search for anything..."
        py="13px"
      />
    </InputGroup>
  );
};
