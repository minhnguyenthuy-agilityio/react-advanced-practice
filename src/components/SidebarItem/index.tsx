import { Link as ChakraLink, Icon, Text } from "@chakra-ui/react";

import { DotIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

import { Link as ReactRouterLink } from "react-router-dom";

interface SidebarItemPropType {
  iconColor: string;
  title: string;
  projectId: string;
}

export const SidebarItem = ({
  iconColor,
  title,
  projectId,
}: SidebarItemPropType) => {
  return (
    <ChakraLink
      as={ReactRouterLink}
      to={`${ROUTES.TASKS_PROJECT_DEFAULT}${projectId}`}
      display="flex"
      w="full"
      pl="3.5"
      py="12.5"
      borderRadius="6"
      alignItems="center"
      _hover={{
        textDecoration: "none",
        bgColor: "secondary.200",
      }}
    >
      <Icon mr="16px" as={DotIcon} color={iconColor} />
      <Text textOverflow="ellipsis" whiteSpace="nowrap" overflowX="hidden">
        {title}
      </Text>
    </ChakraLink>
  );
};
