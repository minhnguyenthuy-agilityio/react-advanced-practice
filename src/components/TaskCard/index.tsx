import { memo } from "react";
import {
  Avatar,
  AvatarGroup,
  Badge,
  Box,
  Center,
  Flex,
  Heading,
  Icon,
  Button,
  Image,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
  Text,
} from "@chakra-ui/react";

import { TASK_STATUS } from "@/constants";

import { TaskInformation, Assignee } from "@/interfaces";

import Gallery from "@/assets/images/gallery.png";

import { FolderIcon, MessageIcon, ThreeDots } from "@/assets/icons";

interface TaskCardProps extends TaskInformation {
  assignees: Assignee[];
  files: number;
  comments: number;
}

export const TaskCard = memo(
  ({
    name,
    priority,
    description,
    image,
    assignees,
    comments,
    files,
  }: TaskCardProps) => {
    return (
      <Box
        as="article"
        maxW="314px"
        minH="177px"
        maxH="328px"
        borderRadius="16px"
        border="solid 1px"
        borderColor="primary.100"
        transition="0.5s"
        p="20px"
        bg="primary.100"
        _hover={{ borderColor: "primary.500" }}
      >
        <Flex justifyContent="space-between">
          <Badge variant={priority}>{priority}</Badge>
          <Menu>
            <MenuButton
              as={Button}
              rightIcon={<ThreeDots />}
              bg="none"
              _hover={{ bg: "none" }}
            />
            <MenuList>
              {TASK_STATUS?.map(({ label }, index) => (
                <MenuItem key={`${label}-${index}`} textTransform="capitalize">
                  {label}
                </MenuItem>
              ))}
            </MenuList>
          </Menu>
        </Flex>

        <Heading
          my="8px"
          size="md"
          textOverflow="ellipsis"
          whiteSpace="nowrap"
          overflowX="hidden"
          overflowY="hidden"
          textTransform="capitalize"
        >
          {name}
        </Heading>

        {image ? (
          <Image
            mb="10px"
            h="110px"
            w="full"
            borderRadius="6px"
            alt="background task"
            objectFit="cover"
            src={image}
          />
        ) : (
          <Center mb="10px" bg="primary.300" borderRadius="10" py="50px">
            <Image
              borderRadius="6px"
              h="40px"
              w="40px"
              alt="background task"
              src={Gallery}
            />
          </Center>
        )}

        <Text
          textOverflow="ellipsis"
          whiteSpace="nowrap"
          overflowX="hidden"
          mb="16px"
          color="secondary.400"
          size="xs"
          textTransform="capitalize"
        >
          {description}
        </Text>

        <Flex mt="8px" justifyContent="space-between" alignItems="center">
          <AvatarGroup spacing="-4px" size="xs" max={5}>
            {assignees?.map(({ avatar, name }, index) => (
              <Avatar key={`${name}-${index}`} src={avatar} />
            ))}
          </AvatarGroup>
          <Flex gap={2}>
            <Flex>
              <Icon as={MessageIcon} />
              <Text ml={1} size="xs">
                {`${comments} ${comments <= 1 ? "comment" : "comments"}`}
              </Text>
            </Flex>
            <Flex>
              <Icon as={FolderIcon} />
              <Text ml={1} size="xs">
                {`${files} ${files <= 1 ? "file" : "files"}`}
              </Text>
            </Flex>
          </Flex>
        </Flex>
      </Box>
    );
  },
);
