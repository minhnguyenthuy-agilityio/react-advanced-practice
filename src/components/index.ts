export * from "./Header";
export * from "./Sidebar";
export * from "./SearchBox";
export * from "./Logo";
export * from "./SidebarItem";
export * from "./TaskCard";
