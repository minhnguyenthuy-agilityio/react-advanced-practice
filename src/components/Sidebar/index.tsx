import { Box, Flex, Icon, Text, VStack } from "@chakra-ui/react";

import { Logo, SidebarItem } from "@/components";

import { PROJECTS } from "@/mocks";

import { TasksIcon } from "@/assets/icons";

export const SideBar = () => {
  return (
    <Box as="aside" h="full" borderRight="1px solid" borderColor="primary.400">
      <Logo />
      <Box pr="2.5" pl="0.5">
        <Flex
          alignItems="center"
          pl="4"
          py="3.5"
          borderBottom="solid 1px "
          borderColor="primary.400"
        >
          <Icon mr="3.5" as={TasksIcon} />
          <Text>Tasks</Text>
        </Flex>

        <VStack as="nav" spacing="2.5" alignItems="start">
          <Text
            pl="3"
            pt="7"
            pb="2"
            textTransform="uppercase"
            size="xs"
            fontWeight="700"
          >
            my projects
          </Text>
          {PROJECTS?.map(({ name, iconColor, projectId }, index) => (
            <SidebarItem
              key={`${name}-${index}`}
              iconColor={iconColor}
              title={name}
              projectId={projectId}
            />
          ))}
        </VStack>
      </Box>
    </Box>
  );
};
