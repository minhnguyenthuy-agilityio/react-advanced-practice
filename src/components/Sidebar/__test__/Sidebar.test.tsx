import { render } from "@/helpers";

import { SideBar } from "@/components";

describe("SideBar component", () => {
  it("matchs snapshoot", () => {
    const element = render(<SideBar />);
    expect(element).toMatchSnapshot();
  });
});
