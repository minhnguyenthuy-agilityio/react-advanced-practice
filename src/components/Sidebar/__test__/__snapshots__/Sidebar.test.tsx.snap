// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component matchs snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <aside
        class="css-xyj4jj"
      >
        <h1
          class="chakra-heading css-2hgecz"
        >
          <a
            class="chakra-link css-1s4iytr"
            href="/"
          >
            <svg
              class="chakra-icon css-zagjba"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-1d9p61o"
            >
              Project M.
            </p>
          </a>
        </h1>
        <div
          class="css-1j2qcn8"
        >
          <div
            class="css-vr1tbx"
          >
            <svg
              class="chakra-icon css-il4tuo"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-80mmfj"
            >
              Tasks
            </p>
          </div>
          <nav
            class="chakra-stack css-152y5d4"
          >
            <p
              class="chakra-text css-199zi8f"
            >
              my projects
            </p>
            <a
              class="chakra-link css-1j5kdpb"
              href="/"
            >
              <svg
                class="chakra-icon css-2xlth4"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  stroke-width="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    stroke-linecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    stroke-linecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    stroke-miterlimit="10"
                  />
                </g>
              </svg>
              <p
                class="chakra-text css-1qf6mdr"
              >
                Mobile App
              </p>
            </a>
            <a
              class="chakra-link css-1j5kdpb"
              href="/pro01"
            >
              <svg
                class="chakra-icon css-5haebz"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  stroke-width="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    stroke-linecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    stroke-linecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    stroke-miterlimit="10"
                  />
                </g>
              </svg>
              <p
                class="chakra-text css-1qf6mdr"
              >
                Website Redesign
              </p>
            </a>
            <a
              class="chakra-link css-1j5kdpb"
              href="/pro02"
            >
              <svg
                class="chakra-icon css-130uht9"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  stroke-width="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    stroke-linecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    stroke-linecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    stroke-miterlimit="10"
                  />
                </g>
              </svg>
              <p
                class="chakra-text css-1qf6mdr"
              >
                Design System
              </p>
            </a>
            <a
              class="chakra-link css-1j5kdpb"
              href="/pro03"
            >
              <svg
                class="chakra-icon css-1k2s7qz"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  stroke-width="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    stroke-linecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    stroke-linecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    stroke-miterlimit="10"
                  />
                </g>
              </svg>
              <p
                class="chakra-text css-1qf6mdr"
              >
                Wireframes
              </p>
            </a>
          </nav>
        </div>
      </aside>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <aside
      class="css-xyj4jj"
    >
      <h1
        class="chakra-heading css-2hgecz"
      >
        <a
          class="chakra-link css-1s4iytr"
          href="/"
        >
          <svg
            class="chakra-icon css-zagjba"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              stroke-width="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                stroke-linecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                stroke-linecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                stroke-miterlimit="10"
              />
            </g>
          </svg>
          <p
            class="chakra-text css-1d9p61o"
          >
            Project M.
          </p>
        </a>
      </h1>
      <div
        class="css-1j2qcn8"
      >
        <div
          class="css-vr1tbx"
        >
          <svg
            class="chakra-icon css-il4tuo"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              stroke="currentColor"
              stroke-width="1.5"
            >
              <path
                d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                fill="none"
                stroke-linecap="round"
              />
              <path
                d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                fill="currentColor"
                stroke-linecap="round"
              />
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11.25"
                stroke-miterlimit="10"
              />
            </g>
          </svg>
          <p
            class="chakra-text css-80mmfj"
          >
            Tasks
          </p>
        </div>
        <nav
          class="chakra-stack css-152y5d4"
        >
          <p
            class="chakra-text css-199zi8f"
          >
            my projects
          </p>
          <a
            class="chakra-link css-1j5kdpb"
            href="/"
          >
            <svg
              class="chakra-icon css-2xlth4"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-1qf6mdr"
            >
              Mobile App
            </p>
          </a>
          <a
            class="chakra-link css-1j5kdpb"
            href="/pro01"
          >
            <svg
              class="chakra-icon css-5haebz"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-1qf6mdr"
            >
              Website Redesign
            </p>
          </a>
          <a
            class="chakra-link css-1j5kdpb"
            href="/pro02"
          >
            <svg
              class="chakra-icon css-130uht9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-1qf6mdr"
            >
              Design System
            </p>
          </a>
          <a
            class="chakra-link css-1j5kdpb"
            href="/pro03"
          >
            <svg
              class="chakra-icon css-1k2s7qz"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <g
                stroke="currentColor"
                stroke-width="1.5"
              >
                <path
                  d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                  fill="none"
                  stroke-linecap="round"
                />
                <path
                  d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                  fill="currentColor"
                  stroke-linecap="round"
                />
                <circle
                  cx="12"
                  cy="12"
                  fill="none"
                  r="11.25"
                  stroke-miterlimit="10"
                />
              </g>
            </svg>
            <p
              class="chakra-text css-1qf6mdr"
            >
              Wireframes
            </p>
          </a>
        </nav>
      </div>
    </aside>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
